File: day1/solution.py
def parse(file_name):
    with open(file_name, "r") as file:
        data = [
            sum([int(x) for x in elf.split()]) 
            for elf in file.read().split("\n\n")
        ]
    return data

def part1(data):
    elf = max(data)
    return elf

def part2(data):
    elfs = sorted(data, reverse=True)
    return sum(elfs[:3])

File: day1/solution.js
const fs = require('fs');

let data;

try {
    data = fs.readFileSync('indata.txt', 'utf8');
  } catch (err) {
    console.error(err);
}

let elfs = data.split("\n\n").map((elf) => elf.split("\n").map((cal) =>  parseInt(cal)).reduce((a,b) => a+b, 0)).sort((a,b) => a-b).reverse()

// Part 1
console.log(elfs[0])

// Part 2
console.log(elfs[0] + elfs[1] + elfs[2])